sdk/face/azure-ai-vision-face/azure/ai/vision/face/_patch.py

```diff
@@ -0,0 +1,965 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Customize generated code here.
+
+Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
+"""
+from typing import Any, IO, List, Optional, Union, cast
+
+from azure.core.tracing.decorator import distributed_trace
+
+from . import models as _models
+from ._client import FaceClient as FaceClientGenerated
+from ._client import FaceSessionClient as FaceSessionClientGenerated
+from ._operations._operations import JSON, _Unset
+
+
+class FaceClient(FaceClientGenerated):
+    """FaceClient.
+
+    :param endpoint: Supported Cognitive Services endpoints (protocol and hostname, for example:
+     https://{resource-name}.cognitiveservices.azure.com). Required.
+    :type endpoint: str
+    :param credential: Credential used to authenticate requests to the service. Is either a
+     AzureKeyCredential type or a TokenCredential type. Required.
+    :type credential: ~azure.core.credentials.AzureKeyCredential or
+     ~azure.core.credentials.TokenCredential
+    :keyword api_version: API Version. Default value is "v1.1-preview.1". Note that overriding this
+     default value may result in unsupported behavior.
+    :paramtype api_version: str or ~azure.ai.vision.face.models.Versions
+    """
+
+    @distributed_trace
+    def detect_from_url(
+        self,
+        body: Union[JSON, IO[bytes]] = _Unset,
+        *,
+        url: str = _Unset,
+        detection_model: Union[str, _models.FaceDetectionModel],
+        recognition_model: Union[str, _models.FaceRecognitionModel],
+        return_face_id: bool,
+        return_face_attributes: Optional[List[Union[str, _models.FaceAttributeType]]] = None,
+        return_face_landmarks: Optional[bool] = None,
+        return_recognition_model: Optional[bool] = None,
+        face_id_time_to_live: Optional[int] = None,
+        **kwargs: Any,
+    ) -> List[_models._models.FaceDetectionResult]:
```

Comment:
I think we need to do a few more things to get this to behave as expected. First let's copy over the overloads from the generated operations here and update them to show our new required keyword-only args: https://github.com/Azure/azure-sdk-for-python/blob/9dc2109e1fb4ec58cb79e8b936566235f3068f04/sdk/face/azure-ai-vision-face/azure/ai/vision/face/_operations/_operations.py#L2532-L2576

Then on the actual implementation, for anything required, I think we need to default to _Unset like you do on `body`/`url`. In this method we'll need to add some logic (like in the generated code shown below) to ensure that any required params are present whether the user passes body or keyword-only arguments.

https://github.com/Azure/azure-sdk-for-python/blob/9dc2109e1fb4ec58cb79e8b936566235f3068f04/sdk/face/azure-ai-vision-face/azure/ai/vision/face/_operations/_operations.py#L2981-L2985

Reply:
`body` can only replace `url` as the other keyword-only required arguments are query parameters. There're not in the request body. Do we need to add these function definitions with `@overload`?

Reply:
Oh I see! I mistakenly thought that `body` included the JSON representation of all of these parameters, didn't realize we had mostly query params here. You can ignore what I said in this comment, sorry for any confusion.